numpy/core/shape_base.py

```diff
@@ -356,3 +361,105 @@ def stack(arrays, axis=0):
     sl = (slice(None),) * axis + (_nx.newaxis,)
     expanded_arrays = [arr[sl] for arr in arrays]
     return _nx.concatenate(expanded_arrays, axis=axis)
+
+
+def block(*arrays):
+    """
+    Create a block array consisting of other arrays.
+
+    You can create a block array with the same notation you use for
+    `np.array`.
```

Comment:
This is Is this completely true? The function only appears to work in 2D, while `np.array` definitely works in ND.